Add a few additional concepts to Configuration and Options articles (#29442)
dotnet/docs: docs/core/extensions/configuration.md

These abstractions are agnostic to their underlying configuration provider (<xref:Microsoft.Extensions.Configuration.IConfigurationProvider>). In other words, you can use an `IConfiguration` instance to access any configuration value from multiple providers.

The binder can use different approaches to process configuration values:​

> - Direct deserialization (using built-in converters) for primitive types​.
> - The <xref:System.ComponentModel.TypeConverter> for complex type when the type has one​.
>   - If not and if the target type has properties, via reflection​.

Reviewer: ```suggestion
- Direct deserialization (using built-in converters) for primitive types​.
- The <xref:System.ComponentModel.TypeConverter> for a complex type when the type has one​.
- Reflection for a complex type that has properties.
```

> [!NOTE]
> The binder has a few limitations:
>

> > - Non-supported properties are ignored that have private setters, or their type cannot be converted.

Reviewer: ```suggestion
> - Properties are ignored if they have private setters or their type can't be converted.
```
